lib/cog/error_response.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Cog.ErrorResponse do
+  alias Piper.Permissions.Ast
+  alias Cog.Models.Bundle
+  alias Cog.Models.Command
+
+  def render({:parse_error, msg}) when is_binary(msg),
+    do: "Whoops! An error occurred. " <> msg
+  def render({:redirect_error, invalid}),
+    do: "Whoops! An error occurred. " <> redirection_error_message(invalid)
+  def render({:binding_error, {:missing_key, var}}),
+    do: "I can't find the variable '$#{var}'."
+  def render({:binding_error, msg}) when is_binary(msg),
+    do: "Whoops! An error occurred. " <> msg
+  def render({:no_rule, current_invocation}),
+    do: "No rules match the supplied invocation of '#{current_invocation}'. Check your args and options, then confirm that the proper rules are in place."
+  def render({:denied, {%Ast.Rule{}=rule, current_invocation}}),
+    do: "Sorry, you aren't allowed to execute '#{current_invocation}' :(\n You will need the '#{rule.permission_selector.perms.value}' permission to run this command."
+  def render({:no_relays, %Bundle{name: name}}),
+    do: "Whoops! An error occurred. No Cog Relays supporting the `#{name}` bundle are currently online"
+  def render({:disabled_bundle, %Bundle{name: name}}),
+    do: "Whoops! An error occurred. The #{name} bundle is currently disabled"
+  def render({:timeout, %Command{}=command}) do
+    name = Command.full_name(command)
+    "The #{name} command timed out"
+  end
+  def render({:template_rendering_error, {error, template, adapter}}),
+    do: "Whoops! An error occurred. There was an error rendering the template '#{template}' for the adapter '#{adapter}': #{inspect error}"
+  def render({:command_error, response}) do
+    error = response.status_message || response.body["message"]
+    "Whoops! An error occurred. " <> error
+  end
+
+  # `errors` is a keyword list of [reason: name] for all bad redirect
+  # destinations that were found. `name` is the value as originally
+  # typed by the user.
+  defp redirection_error_message(errors) do
```

Comment:
@christophermaier Putting this in a template felt weird because it's really only part of the final template rendered. I opted to leave it here for now.